examples/pigweed-app/nrfconnect/prj.conf

```diff
@@ -0,0 +1,87 @@
+#
+#    Copyright (c) 2020 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+CONFIG_CHIP=y
+
+CONFIG_NEWLIB_LIBC=y
+
+# Export POSIX names for functions implementing a subset of POSIX standard in Zephyr
+CONFIG_POSIX_API=y
+CONFIG_PTHREAD_IPC=y
+CONFIG_EVENTFD=y
+
+CONFIG_LOG=y
+CONFIG_LOG_MINIMAL=y
+CONFIG_ASSERT=y
+CONFIG_HW_STACK_PROTECTION=y
+
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=n
+
+# Application stack size
+CONFIG_MAIN_STACK_SIZE=8192
+CONFIG_INIT_STACKS=y
+
+# Disable certain parts of Zephyr IPv6 stack
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+# Use mbedTLS from nrf_security library
+CONFIG_NORDIC_SECURITY_BACKEND=y
+CONFIG_OPENTHREAD_MBEDTLS=n
+
+CONFIG_MBEDTLS_ENABLE_HEAP=y
+CONFIG_MBEDTLS_HEAP_SIZE=15360
+CONFIG_MBEDTLS_TLS_LIBRARY=y
+CONFIG_NRF_SECURITY_ADVANCED=y
+
+CONFIG_MBEDTLS_AES_C=y
+CONFIG_MBEDTLS_RSA_C=y
+CONFIG_MBEDTLS_ECP_C=y
+CONFIG_MBEDTLS_ECP_DP_SECP256R1_ENABLED=y
+CONFIG_MBEDTLS_CTR_DRBG_C=y
+CONFIG_MBEDTLS_CIPHER_MODE_CTR=y
+CONFIG_MBEDTLS_ECJPAKE_C=y
+
+# Disable unneeded crypto operations
+CONFIG_MBEDTLS_SHA512_C=n
+CONFIG_MBEDTLS_CIPHER_MODE_XTS=n
+CONFIG_MBEDTLS_CHACHA20_C=n
+CONFIG_MBEDTLS_POLY1305_C=n
+CONFIG_MBEDTLS_CHACHAPOLY_C=n
+CONFIG_MBEDTLS_GCM_C=n
+
+# Add support for LEDs and buttons on Nordic development kits
+CONFIG_DK_LIBRARY=y
+
+# Long log strings
+CONFIG_LOG_STRDUP_MAX_STRING=256
+CONFIG_LOG_STRDUP_BUF_COUNT=20
+
+# Additional configs for debbugging experience.
+CONFIG_THREAD_NAME=y
+CONFIG_MPU_STACK_GUARD=y
+
+# Enable Pigweed RPC
+CONFIG_CHIP_PW_RPC=y
```

Comment:
I guess all the previous settings are already in the common config, so you don't have to repeat them. Only make sure that you set CONFIG_STD_CPP11=n and CONFIG_OPENTHREAD_SHELL=n.

Reply:
I know, but I didn't use common config here, as I thought that it is not necessary to include whole network stack and OpenThread, as app for now doesn't need any of this things.

Reply:
Ok, I guess we may later think about finer granularity of the common configs.